### ISSUE TITLE
Recommendation for unnamed arguments

### ISSUE THREAD

**Issue by mslinn** [acrisci/commander-scala#1]
If a program has several unnamed arguments, how would you suggest parsing them? For example, the case class below stores 5 options and 4 unnamed parameters (with defaults) that configure a program. If values for the 4 unnamed parameters are supplied they should be used, in the order supplied. Otherwise, default values should be used.

Looking at the source code, it seems that the values not associated with any option are passed along as `program.args`. Is that the best way to access those values for this purpose? Some of those values might need to be converted to Ints, Longs, etc, so it would be nice to use the mechanism already in `commander-scala` for doing that.

```
import com.github.acrisci.commander.Program
import java.io.File

abstract sealed case class Options(
  consumerKey: String,
  consumerSecret: String,
  accessToken: String,
  accessTokenSecret: String,
  overWrite: Boolean = false,
  tweetDirectory: File = new File("~/twitterClassifier/tweets/"),
  numTweetsToCollect: Int = 100,
  intervalSecs: Int = 1,
  partitionsEachInterval: Int = 1
)

object Options {
  def apply(args: Array[String]): Options = {
    val program = new Program()
      .version("2.0.0")
      .option(flags="-c, --consumerKey [type]",       description="Twitter OAuth Consumer Key",        required=true)
      .option(flags="-s, --consumerSecret [type]",    description="Twitter OAuth Consumer Secret",     required=true)
      .option(flags="-t, --accessToken [type]",       description="Twitter OAuth Access Token",        required=true)
      .option(flags="-v, --accessTokenSecret [type]", description="Twitter OAuth Access Token Secret", required=true)
      .option(flags="-x, --overWrite",                description="Overwrite data files on each run [false]", default=false)
      .parse(args)

    if (args.isEmpty) program.help

    new Options(
      consumerKey = program.consumerKey,
      consumerSecret = program.consumerSecret,
      accessToken = program.accessToken,
      accessTokenSecret = program.accessTokenSecret,
      overWrite = program.overWrite
     // mandatory unnamed parameters would go here
    ){}
  }
}
```

**Comment by acrisci** [comment]
Sorry for the late reply.

At a glance it seems this feature is not implemented in the project this is based on, [commander.js](https://github.com/tj/commander.js) but it's a very good feature.

I really like the way the python argparse module does this.

``` python
parser = argparse.ArgumentParser(description='Process some integers.')
parser.add_argument('integers', metavar='N', type=int, nargs='+',
                    help='an integer for the accumulator')
```

So in this case, without the `--` in front of the option name, it is parsed as a positional argument.

In commander-scala, this could perhaps be implemented the same way.

``` scala
val program = new Program()
      .version("2.0.0")
      .option("some-positional-arg",  description="A positional argument", required=true, metavar="ARG")
```

So there, the `flags` would not have the `--` in front of it (kind of not making it a flag anymore, which is a problem) and then that would be a positional argument which can be accessed on the `program`.

```
$ ./run-program 8
program.somePositionalArgument[Int]
> 8
```

If you'd like to implement something like that, I would accept it into the project.

**Comment by mslinn** [comment]
I did not want to change the syntax of the app I was working on, so I simply referenced `program.args`. You might want to mention it in the documentation.

BTW, I wrote up a brief [blog post](https://medium.com/@mslinn/lessons-from-updating-the-twitter-classifier-apache-spark-reference-application-3c841bd69c0b#.mx9fgxidt) of how I used Commander Scala to replace Apache Commons CLI in the new version of the Apache Spark Twitter Classifier Reference Application.
